homeassistant/components/airthings_ble/sensor.py

```diff
@@ -183,4 +184,12 @@ def __init__(
     @property
     def native_value(self) -> StateType:
         """Return the value reported by the sensor."""
-        return self.coordinator.data.sensors[self.entity_description.key]
+        current_value = self.coordinator.data.sensors.get(self.entity_description.key)
+
+        if current_value:
+            self._attr_assumed_state = False
+            self._last_valid_value = current_value
+        else:
+            self._attr_assumed_state = True
+
+        return self._last_valid_value
```

Comment:
This should be implemented in `_handle_coordinator_update` instead.  We don't want properties to have side effects

Reply:
So essentially moving the same concept to `_handle_coordinator_update`?

Reply:
Yes, but I think we should eventually mark it as unavailable after it fails X amount of times since at that point we know something is wrong.

I can't suggest a value for X though as I don't have one of these devices to know

Reply:
Been thinking about a timeout myself too, and since it fetches every 5 minutes I would be ok with 5 more attempts (30 mins) given that a lot can happen during this time in the space that the device is within.

Reply:
So I'm going to advocate for the opposite: when I see high unavailability now it indicates (from my experience) that there is a hardware problem, low battery, or poor bluetooth coverage or interference that needs fixing (i.e. add an esphome proxy). If the integration starts holding over "last known" data then it will hide indications of such issues.